kafka-1.0.x/src/main/scala/monix/kafka/CommittableOffset.scala

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (c) 2014-2019 by The Monix Project Developers.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package monix.kafka
+
+import monix.eval.Task
+import org.apache.kafka.common.TopicPartition
+
+/** Represents offset for specified topic and partition that can be
+  * committed by [[commit]] method call.
+  * To achieve good performance it is recommended to use batched commit with
+  * [[CommittableOffsetBatch]] class.
+  *
+  * @param topicPartition is the topic and partition identifier
+  *
+  * @param offset is the offset to be committed
+  *
+  * @param commitBatch is the function for batched commit realized as closure
+  *        in [[KafkaConsumerObservable]] context. This decision was made for
+  *        thread-safety reasons.
+  */
+final class CommittableOffset(
+  val topicPartition: TopicPartition,
+  val offset: Long,
+  private[kafka] val commitBatch: Map[TopicPartition, Long] => Task[Unit]) {
+
+  /**
+    * Commits [[offset]] for the [[topicPartition]] to Kafka. It is recommended
+    * to use batched commit with [[CommittableOffsetBatch]] class.
+    * */
+  def commit(): Task[Unit] = commitBatch(Map(topicPartition -> offset))
+}
+
+object CommittableOffset {
+
+  def apply(
+    topicPartition: TopicPartition,
+    offset: Long,
+    commitBatch: Map[TopicPartition, Long] => Task[Unit]): CommittableOffset =
```

Comment:
I feel like we shouldn't expose this in constructor, we could make a version similar to `CommittableOffsetBatch` with default dummy `commitBatch`

Reply:
I did not understand this point. Can you please explain it with code snippet?

I think that `CommittableOffset` with dummy `commitBatch` does not make sense. `CommittableOffsetBatch` has dummy callback because _batch_ can be empty, but _single_ offset can not.

Reply:
How are users supposed to create `commitBatch` function? 
I guess they can if they have access to `KafkaConsumer` but it seems error prone

Reply:
Oh, I forgot to make `apply` function private. Fixed now.